### PR TITLE
resolves #2266

### PR DESCRIPTION
**Pull request by n2aws**
Small edit to make the fact page of the API honor the view_facts permission

### CODE REVIEW

File: lib/foreman/access_permissions.rb [hunk 1]
```diff
@@ -252,7 +252,8 @@
   end
 
   map.security_block :facts do |map|
-    map.permission :view_facts,       {:fact_values => [:index, :show, :auto_complete_search]}
+    map.permission :view_facts,
+                   :facts => [:index, :show], :"api/v1/fact_values" => [:index, :show]
```
Comment (skottler) [inline lib/foreman/access_permissions.rb:256]: This will cause breakage going forward with the versioned API.

**Comment by ohadlevy** [comment]
@skottler why?

**Comment by isratrade** [comment]
@skottler, @palmerit what about adding :"api/v1/fact_values" => and :"api/v2/fact_values" => in the meantime.  It's not the greatest thing to maintain for future versions, but it could be re-factored later.

**Comment by domcleal** [comment]
I'd suggest adding it for now, we already have a few other mentions of API v1 in the file.  I'd like to investigate a more sustainable way in #2248, it can be changed then if we have a solution.

**Comment by n2aws** [comment]
I just duplicated the same format the others used.  So,  if this patch
isn't compatible.. I'd venture a guess that none of the current API
permissions pieces are compatible either.

Either way,  I'll retract the patch.   I've created the issue, and you guys
can figure out the best way to implement it since I'm not in-the-loop as to
what v2 entails.

**Comment by ohadlevy** [comment]
rebased and merged - thanks @palmerit

**Pull request closed without merging**